src/Core/FakeVimCommand.cpp

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (C) 2019-2020 Ashar Khan <ashar786khan@gmail.com>
+ *
+ * This file is part of CP Editor.
+ *
+ * CP Editor is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * I will not be responsible if CP Editor behaves in unexpected way and
+ * causes your ratings to go down and or lose any important contest.
+ *
+ * Believe Software is "Software" and it isn't immune to bugs.
+ *
+ */
+#include "Core/FakeVimCommand.hpp"
+#include "Util/FileUtil.hpp"
+#include "Widgets/TestCases.hpp"
+#include "appwindow.hpp"
+#include "fakevimhandler.h"
+#include "generated/SettingsHelper.hpp"
+#include <QApplication>
+#include <QFileInfo>
+#include <QStatusBar>
+
+namespace Core
+{
+FakeVimCommand::FakeVimCommand(AppWindow *aw) : appwin(aw)
+{
+}
+
+bool FakeVimCommand::handle(FakeVim::Internal::ExCommand const &cmd)
```

Comment:
This function handles the custom commands. It calls many helper methods like `wantSomething(cmd)` which returns true. If command `cmd` is requesting to do what the function names say. In some cases like `open` the function returns a `QPair<bool, QString>`, bool indicates that `cmd` matched with the action represented by function name, and if it is true, the QString contains the arguments to the command. The arguments are verified beforehand, so in this case, the open ensures that file to open exists and is a c/c++/java/py source. If the argument failed the checks, an empty string is returned and the `wantSomething(cmd)` has warned the user of why it failed to return the argument.